src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,65 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.dto.MovieSessionDto;
+import mate.academy.spring.mapper.MovieSessionMapper;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.service.MovieSessionService;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private static final String DATE_FORMAT_TO_PARSE = "dd.MM.yyyy";
+    private final MovieSessionMapper movieSessionMapper;
+    private final MovieSessionService movieSessionService;
+
+    public MovieSessionController(MovieSessionMapper movieSessionMapper,
+                                  MovieSessionService movieSessionService) {
+        this.movieSessionMapper = movieSessionMapper;
+        this.movieSessionService = movieSessionService;
+    }
+
+    @PostMapping
+    public MovieSessionDto create(@RequestBody MovieSessionDto movieSessionDto) {
+        return movieSessionMapper
+                .toDto(movieSessionService.add(movieSessionMapper.toModel(movieSessionDto)));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionDto> getAllAvailableSessions(@RequestParam Long movieId,
+                                                         @RequestParam
+                                                         @DateTimeFormat(pattern =
+                                                                 DATE_FORMAT_TO_PARSE)
+                                                         LocalDate date) {
```

Comment:
HINT: You also can write code in a such way (if checkstyle allow it)
```suggestion
    public List<MovieSessionDto> getAllAvailableSessions(
    	@RequestParam Long movieId,
        @RequestParam
        @DateTimeFormat(pattern = DATE_FORMAT_TO_PARSE) LocalDate date
    ) {
```